MapboxCoreNavigation/Router.swift

```diff
@@ -12,6 +12,11 @@ public protocol RouterDataSource: class {
     var locationProvider: NavigationLocationManager.Type { get }
 }
 
+/**
+ A route and its index in a `RouteResponse` that sorts routes from most optimal to least optimal.
+ */
+public typealias IndexedRoute = (Route, Int)
```

Comment:
This is the most generic place I could find for a type that ends up getting used all over the place. The “indexed” pattern is inspired by Turf for Swift. The idea is to force code that sets the route to a new value to also set the route index to a potentially new value at the same time, avoiding stale state or race conditions.